les/peer.go

```diff
@@ -605,7 +605,8 @@ func (p *serverPeer) Handshake(genesis common.Hash) error {
 		if recv.get("txRelay", nil) != nil {
 			p.onlyAnnounce = true
 		}
-		p.serveTxLookup = recv.get("disableTxLookup", nil) != nil
+		var recentTx uint
+		p.serveTxLookup = recv.get("recentTxLookup", &recentTx) != nil || recentTx == 0
```

Comment:
There are two cases: 
- The recentTxLookup is not included. 
   - If the version is less than v4, accept
   - If the version is >= v4, reject
- The recentTxLookup is included
   - If the peer version < v4, reject
   - If the peer version >= v4
       - If the recentTx is 0, enable tx serving
       - If the recentTx is non-0, disable tx serving

Reply:
Right, I did not check the protocol version at all. Now I always expect the field if the protocol is at least lpv4 and fail if it is missing. If it's an old protocol version then I don't check it at all since it has no defined meaning there.